module/VuFind/src/VuFind/Search/Solr/Results.php

```diff
@@ -265,11 +267,11 @@ protected function performSearch()
     /**
      * Try to fix a query that caused a parser error.
      *
-     * @param AbstractQuery $query Bad query
+     * @param QueryInterface $query Bad query
      *
-     * @return bool|AbstractQuery  Fixed query, or false if no solution is found.
+     * @return bool|QueryInterface  Fixed query, or false if no solution is found.
      */
-    protected function fixBadQuery(AbstractQuery $query)
+    protected function fixBadQuery(QueryInterface $query): bool|QueryInterface
```

Comment:
Is it worth considering switching this to `?QueryInterface` while we're changing things anyway? Using null generally feels cleaner than using false. I haven't evaluated what the impact would be, though I would expect it to be minimal. We could deal with it in a follow-up PR if you think it's worth doing but better kept separate... or we could just leave it alone if that's best.